go_modules/lib/dependabot/go_modules/update_checker.rb

```diff
@@ -16,7 +16,8 @@ class UpdateChecker < Dependabot::UpdateCheckers::Base
         /no go-import meta tags/,
         # Package url 404s
         /404 Not Found/,
-        /Repository not found/
+        /Repository not found/,
+        /unrecognized import path/
```

Comment:
Xref https://github.com/dependabot/dependabot-core/pull/2708 , cc @feelepxyz 

This was part of the previous implementation, but we split the two expected messages (something like: `unrecognized import path: Repository not found` and  `unrecognized import path: no go-import meta tags`) into separate cases.

The goal of this PR is addressing an unexpected message: `unrecognized import path "system-tests": import path does not begin with hostname`.

This could alternatively use that message,`import path does not begin with hostname`. My thinking was this fail-safe made sense, and the ordering of the array meant the more specific messages would still be matched first.

The change to `FileParser#skip_dependency?` also means the "import path does not begin with hostname" message is no longer expected - such modules are filtered and not checked.